### PR TITLE
fix: Currency input widget round off value if there is a trailing 0 a…

### LINKED ISSUE #12297: [Bug]: Currency Input rounds off text  if there is a trailing zero
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When a user enters `11.90` on the Currency widget and clicks outside, the text gets rounded to `12`.

### Steps To Reproduce

1. Drop a currency widget on the Canvas
2. type `11.90`
3. Click outside the widget.
4. Observe that text gets rounded to `12`.

### Public Sample App

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by sbalaji1192**
…fter decimal

Fixes #12297

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/currency-input-round-off 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.68 **(-0.01)** | 37.06 **(0)** | 35.25 **(0)** | 55.94 **(-0.01)**
 :red_circle: | app/client/src/widgets/CurrencyInputWidget/component/utilities.ts | 96.88 **(-0.18)** | 70.83 **(0)** | 100 **(0)** | 96 **(-0.3)**
 :red_circle: | app/client/src/widgets/CurrencyInputWidget/widget/index.tsx | 46.3 **(-0.43)** | 20 **(0)** | 40 **(0)** | 46.23 **(-0.44)**</details>

### CODE REVIEW

File: app/client/src/widgets/CurrencyInputWidget/widget/index.tsx [hunk 1]
```diff
@@ -253,18 +254,17 @@ class CurrencyInputWidget extends BaseInputWidget<
   handleFocusChange = (isFocused?: boolean) => {
     try {
       if (isFocused) {
-        const deFormattedValue = parseLocaleFormattedStringToNumber(
-          this.props.text,
-        );
-        this.props.updateWidgetMetaProperty(
-          "text",
-          isNaN(deFormattedValue) ? "" : String(deFormattedValue),
+        const text = this.props.text || "";
```
Comment (sbalaji1192) [inline app/client/src/widgets/CurrencyInputWidget/widget/index.tsx:257]: we are only removing the thousand separators so we don't loose the trailing zero after decimal

File: app/client/src/widgets/CurrencyInputWidget/widget/index.tsx [hunk 2]
```diff
@@ -253,18 +254,17 @@ class CurrencyInputWidget extends BaseInputWidget<
...
+          new RegExp("\\" + getLocaleThousandSeparator(), "g"),
+          "",
         );
+        this.props.updateWidgetMetaProperty("text", deFormattedValue);
       } else {
         if (this.props.text) {
           const formattedValue = formatCurrencyNumber(
             this.props.decimals,
-            String(this.props.value),
+            this.props.text,
```
Comment (sbalaji1192) [inline app/client/src/widgets/CurrencyInputWidget/widget/index.tsx:267]: text preserves the trailing zeros after decimals since its a string

**Review by ashit-rath — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@sbalaji1192 Tested this PR and it is working as expected

**Review by ashit-rath — APPROVED** [review]

**Comment by diepeterpan** [comment]
Maybe I am doing something wrong, I have a currency widget set to two decimals but it rounds to the nearest whole number when I move off the field. **Appsmith Community v1.8.8 - Released 4 days ago**.  What am I doing wrong?

image
image

**Pull request merged**